Bottom: 6b041962f24e4bafc0defcb20f7343086bd5abff
Top:    a44ab6c4812ce484728b20b3f304c197d8f43a6d
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-02-07 17:10:53 -0600

Make bringup.sh more robust

Fix the parameter expansion of $1 with default.

Suppress output when adjusting the known_hosts file, in case the host
wasn't there to begin with.

Any errors adjusting known_hosts would cause the script to exit, so move the
set command so only ansible errors will exit.


---

diff --git a/ansible-playbook/bringup.sh b/ansible-playbook/bringup.sh
index cdc89f5..d180baf 100755
--- a/ansible-playbook/bringup.sh
+++ b/ansible-playbook/bringup.sh
@@ -1,14 +1,15 @@
 #!/bin/bash
-set -e
 
-inventory_file=${1:staging}
+inventory_file=${1:-staging}
 
 for host in $(sed -e 's|^[[].*$||' -e 's|\..*$||' -e '/^$/d' ${inventory_file} | sort | uniq)
 do
-  ssh-keygen -R ${host}.novatech-llc.com
-  ssh-keygen -R ${host}
-  ssh-keygen -R $(getent hosts ${host} | awk '{ print $1 ; exit }')
+  ssh-keygen -R ${host}.novatech-llc.com 1>/dev/null 2>&1
+  ssh-keygen -R ${host} 1>/dev/null 2>&1
+  ssh-keygen -R $(getent hosts ${host} | awk '{ print $1 ; exit }') 1>/dev/null 2>&1
   ssh-keyscan ${host}.novatech-llc.com >> ~/.ssh/known_hosts
 done
+
+set -e
 ansible-playbook bootstrap.yml -i ${inventory_file} --ask-pass
 ansible-playbook site.yml -i ${inventory_file}
